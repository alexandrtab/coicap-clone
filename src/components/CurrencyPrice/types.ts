export interface ICurrencyPriceProps {
	currencySymbol: string;
	priceSize: number;
	priceColor: string;
	price: string;
}
