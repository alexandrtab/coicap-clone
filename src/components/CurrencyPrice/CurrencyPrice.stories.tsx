import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { CurrencyPrice } from "./CurrencyPrice";
import "../../app/styles/globals.scss";

export default {
	title: "Currency price",
	component: CurrencyPrice,
	argTypes: {
		currencySymbol: {
			type: "string",
			description: "Символ валюты",
			defaultValue: "$"
		},
		priceSize: {
			type: "number",
			description: "Размер текста цены валюты",
			defaultValue: "1",
		},
		priceColor: {
			type: "string",
			description: "Цвет текста цены валюты",
			defaultValue: "black",
			control: {
				type: "color",
			}
		},
		price: {
			type: "string",
			description: "Цена валюты",
			defaultValue: "23853.39",
			control: {
				type: "number",
			}
		},
	}
} as ComponentMeta<typeof CurrencyPrice>;

const Template: ComponentStory<typeof CurrencyPrice> = (args) => <CurrencyPrice { ...args } />;

export const Primary = Template.bind({});

Primary.args = {
	currencySymbol: "$",
	priceSize: 1.2,
	priceColor: "black",
	price: "23853.39",
};
