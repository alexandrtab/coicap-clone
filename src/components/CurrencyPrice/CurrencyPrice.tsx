import React from "react";

import { ICurrencyPriceProps } from "./types";

export const CurrencyPrice: React.FC<ICurrencyPriceProps> = ({
	currencySymbol,
	price,
	priceSize,
	priceColor,
}) => {
	return (
		<div data-testid="currency-price-block" style={ { display: "flex", justifyContent: "center", fontSize: `${priceSize}rem`, } }>
			<span data-testid="currency-symbol" className="currency-symbol">{currencySymbol}</span>
			<p data-testid="currency-price" style={ { color: priceColor, marginLeft: "5px" } }>{price}</p>
		</div>
	);
};
