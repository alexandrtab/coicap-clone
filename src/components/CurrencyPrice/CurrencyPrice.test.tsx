import { render, screen } from "@testing-library/react";

import { CurrencyPrice } from "./CurrencyPrice";
import "@testing-library/jest-dom";

describe("CurrencyPrice", () => {
	render(
		<CurrencyPrice currencySymbol={ "$" } priceSize={ 1 } priceColor={ "green" } price={ "" }/>
	);
	test("should render correct with all props and be equal to snapshot", () => {
		const divElem = screen.getByTestId("currency-price-block");
		const symbol = screen.getByTestId("currency-symbol");
		const price = screen.getByTestId("currency-price");

		expect(divElem).toHaveStyle({ display: "flex", justifyContent: "center", fontSize: "1rem" });
		expect(price).toHaveStyle({ color: "green", marginLeft: "5px" });
		expect(symbol).toHaveTextContent("$");
		expect(divElem).toMatchSnapshot();
		expect(price).toMatchSnapshot();
		expect(symbol).toMatchSnapshot();
	});
});
