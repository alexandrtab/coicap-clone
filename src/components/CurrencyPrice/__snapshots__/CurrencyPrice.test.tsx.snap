// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyPrice should render correct with all props 1`] = `
<span
  class="currency-symbol"
  data-testid="currency-symbol"
>
  $
</span>
`;

exports[`CurrencyPrice should render correct with all props 2`] = `
<span
  class="currency-symbol"
  data-testid="currency-symbol"
>
  $
</span>
`;

exports[`CurrencyPrice should render correct with all props 3`] = `
<span
  class="currency-symbol"
  data-testid="currency-symbol"
>
  $
</span>
`;

exports[`CurrencyPrice should render correct with all props and be  1`] = `
<div
  data-testid="currency-price-block"
  style="display: flex; justify-content: center; font-size: 1rem;"
>
  <span
    class="currency-symbol"
    data-testid="currency-symbol"
  >
    $
  </span>
  <p
    data-testid="currency-price"
    style="color: green; margin-left: 5px;"
  />
</div>
`;

exports[`CurrencyPrice should render correct with all props and be  2`] = `
<p
  data-testid="currency-price"
  style="color: green; margin-left: 5px;"
/>
`;

exports[`CurrencyPrice should render correct with all props and be  3`] = `
<span
  class="currency-symbol"
  data-testid="currency-symbol"
>
  $
</span>
`;

exports[`CurrencyPrice should render correct with all props and be equal to snapshot 1`] = `
<div
  data-testid="currency-price-block"
  style="display: flex; justify-content: center; font-size: 1rem;"
>
  <span
    class="currency-symbol"
    data-testid="currency-symbol"
  >
    $
  </span>
  <p
    data-testid="currency-price"
    style="color: green; margin-left: 5px;"
  />
</div>
`;

exports[`CurrencyPrice should render correct with all props and be equal to snapshot 2`] = `
<p
  data-testid="currency-price"
  style="color: green; margin-left: 5px;"
/>
`;

exports[`CurrencyPrice should render correct with all props and be equal to snapshot 3`] = `
<span
  class="currency-symbol"
  data-testid="currency-symbol"
>
  $
</span>
`;
